obelix/shon_min.py

```diff
@@ -0,0 +1,156 @@
+    m_range = re.match(r"^\s*(-?\d+(?:\.\d+)?)\s*-\s*(-?\d+(?:\.\d+)?)\s*$", s)
+    if m_range:
+        a, b = float(m_range.group(1)), float(m_range.group(2))
+        if a == 10:
+            return 10 ** (-b)
+        return (a + b) / 2.0
+
+    # Standard sci notation "coef × 10 - exp"
+    m_sci = re.match(r"^\s*(?P<coef>-?\d+(?:\.\d+)?)?\s*[×x*]?\s*10\s*-\s*(?P<exp>\d+)\s*$", s)
+    if m_sci:
+        coef = float(m_sci.group("coef")) if m_sci.group("coef") not in (None, "") else 1.0
+        exp = int(m_sci.group("exp"))
+        return coef * 10 ** (-exp)
+
+    # Search anywhere for sci pattern
+    m_any = re.search(r"(?P<coef>-?\d+(?:\.\d+)?)?\s*[×x*]?\s*10\s*-\s*(?P<exp>\d+)", s)
+    if m_any:
+        coef = float(m_any.group("coef")) if m_any.group("coef") not in (None, "") else 1.0
+        exp = int(m_any.group("exp"))
+        return coef * 10 ** (-exp)
+
+    # Fallback to float
+    try:
+        return float(s)
+    except ValueError:
+        return np.nan
+
+# Unit conversion
+UNIT_CONVERSION = {
+    "S/cm":    1,
+    "Scm-1":   1,
+    "mScm−1":  1e-3,
+    "mScm-1":  1e-3,
+    "mS/cm":   1e-3
+}
+
+def convert_to_S_cm(unit: str, value: float) -> float:
+    """Convert given value into S/cm based on its raw unit."""
+    factor = UNIT_CONVERSION.get(unit, None)
+    if factor is None:
+        return value
+    return value * factor
+
+# Formula validation
+def is_valid_formula(formula: str) -> bool:
+    """Return True if pymatgen can parse the formula into a non-empty composition."""
+    try:
+        comp = Composition(formula)
+        return len(comp.get_el_amt_dict()) > 0
+    except Exception:
+        return False
+
+# Main processing pipeline
+def main(input_xlsx: str = "ao3c01424_si_001.xlsx",
+         output_xlsx: str = "Shon_and_Min.xlsx"):
+    # 1) Load raw data
+    df = pd.read_excel(input_xlsx, sheet_name="Sheet2")
+
+    # 2) Parse ionic conductivity strings
+    df["Ionic Conductivity Numeric"] = df["Ionic Conductivity"].apply(convert_scientific_string)
+
+    # 3) Convert to standard units (S/cm)
+    df["Ionic Conductivity Numeric (S/cm)"] = df.apply(
+        lambda r: convert_to_S_cm(r.get("Raw_unit", ""), r["Ionic Conductivity Numeric"]),
+        axis=1
+    )
+
+    # 4) Validate & filter rows
+    to_drop = []
+    for idx, row in df.iterrows():
+        cond = row["Ionic Conductivity Numeric (S/cm)"]
+        # must be positive and log10 in [-18, 0]
+        if pd.isna(cond) or cond <= 0 or not (-18 <= np.log10(cond) <= 0):
+            to_drop.append(idx)
+            continue
+        # must have a valid formula
+        name = row.get("Name", "")
+        if pd.isna(name) or not is_valid_formula(name):
+            to_drop.append(idx)
+            continue
+
+    df_clean = df.drop(index=to_drop).reset_index(drop=True)
+
+    # 5) Compute log10 target
+    df_clean["log_target"] = np.log10(df_clean["Ionic Conductivity Numeric (S/cm)"])
```

Comment:
I would not apply log10 here because we don't apply it for other dataset, we can let users decide to do that